src/parser/xasl_generation.c

```diff
@@ -18460,6 +18460,49 @@ pt_to_constraint_pred (PARSER_CONTEXT * parser, XASL_NODE * xasl, PT_NODE * spec
 
 }
 
+static XASL_NODE *
+pt_to_xasl_for_dblink (PARSER_CONTEXT * parser, PT_NODE * spec)
+{
+  assert (parser != NULL && spec != NULL);
+
+  XASL_NODE *xasl = regu_xasl_node_alloc (INSERT_PROC);
+  if (xasl == NULL)
+    {
+      return NULL;
+    }
+
+  PT_NODE *server_spec = spec->info.spec.remote_server_name;
+  PT_DBLINK_INFO *pdblink = &(server_spec->info.dblink_table);
+
+  assert (server_spec->node_type == PT_DBLINK_TABLE_DML);
```

Comment:
The query below throws an assert because PT_DBLINK_TABLE_DML is not.
```
create synonym s1 for t1@cubrid_dblink;
insert into s1 values (1);

csql: src/parser/xasl_generation.c:18477: XASL_NODE* pt_to_xasl_for_dblink(PARSER_CONTEXT*, PT_NODE*): Assertion `server_spec->node_type == PT_DBLINK_TABLE_DML' failed.
Aborted (core dumped)
```

Reply:
error is from create synonym s1 for t1@cubrid_dblink;

Reply:
Same below.
```
insert into t1@cubrid_dblink values (1);
```
I made a synonym like below for testing.
```
create synonym s1 for dba.t1@cubrid_dblink;
```

Reply:
This is resolved by next commit.